src/main/java/com/ericsson/esj/JenkinsManager.java

```diff
@@ -0,0 +1,259 @@
+
+    private static final String JENKINS_JOB_TEMPLATE_FILE_PATH = String.join(File.separator, "src", "main",
+            "resources", "jenkinsJobTemplate.xml");
+    private static final String DEFAULT_JOB_TOKEN = "test";
+    private static final String DEFAULT_SCRIPT = "echo &quot;Test&quot;";
+    private static final String DEFAULT_JOB_NAME = "test_job";
+
+    private String jenkinsBaseUrl;
+    private String encoding;
+    private String crumb;
+
+    /**
+     * Constructor, takes jenkins baseUrl, username and password.
+     * @param jenkinsBaseUrl
+     * @param username
+     * @param password   **  Jenkins password or API-token
+     * @throws UnsupportedEncodingException
+     * @throws URISyntaxException
+     * @throws JSONException
+     */
+    public JenkinsManager(String jenkinsBaseUrl, String username, String password)
+            throws UnsupportedEncodingException, URISyntaxException, JSONException {
+        String authString = String.join(":", username, password);
+
+        this.jenkinsBaseUrl = jenkinsBaseUrl;
+        this.encoding = DatatypeConverter.printBase64Binary(authString.getBytes("utf-8"));
+        this.crumb = getCrumb();
+    }
+
+    /**
+     * Constructor, takes jenkins protocol, host, port, username and password.
+     * @param protocol
+     * @param host
+     * @param port
+     * @param username
+     * @param password   **  Jenkins password or API-token
+     * @throws UnsupportedEncodingException
+     * @throws URISyntaxException
+     * @throws JSONException
+     */
+    public JenkinsManager(String protocol, String host, int port, String username, String password)
+            throws UnsupportedEncodingException, URISyntaxException, JSONException {
+        String authString = String.join(":", username, password);
+        this.jenkinsBaseUrl = String.format("%s://%s:%d",protocol, host, port);
+        this.encoding = DatatypeConverter.printBase64Binary(authString.getBytes("utf-8"));
+        this.crumb = getCrumb();
+    }
+
+    /**
+     * Takes a token and script and returns XML data for jenkins with job_token and
+     * script added.
+     *
+     * @param job_token
+     * @param script
+     * @return
+     * @throws IOException
+     */
+    public String getXmlJobData(String job_token, String script) throws IOException {
+        String jobData = getStringFromFile(JENKINS_JOB_TEMPLATE_FILE_PATH);
+
+        if (job_token == null) {
+            job_token = DEFAULT_JOB_TOKEN;
+        }
+        if (script == null) {
+            script = DEFAULT_SCRIPT;
+        }
+
+        jobData = injectScriptAndTokenIntoXmlString(jobData, job_token, script);
+        return jobData;
+    }
+
+    /**
+     * Creates a jenkins job with a given name using the XML data as input for
+     * job configuration
+     *
+     * @param jobName
+     * @param jobXmlData
+     * @return
+     * @throws URISyntaxException
+     * @throws ClientProtocolException
```

Comment:
It does not look like it throws all these exception

Reply:
It does now :)